contributors/design-proposals/federated-statefulsets.md

```diff
@@ -0,0 +1,248 @@
+# Federated StatefulSet design
+irfan.rehman@huawei.com, dhilip.kumar.s@huawei.com
+
+# Goal
+
+To outline the use cases for the need of federated Statefulset and determine suitable design/implementation approach for the same.
+
+# Background
+
+The original design proposal of the statefulset in kubernetes, including the use cases and example applications, can be found [here](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/stateful-apps.md).
+
+
+# Use Cases
+
+1 – A stateful app, for the reasons of high availability, wants the stateful pods distributed in different clusters, such that the set can withstand cluster failures. This represents an app with one single global quorum.
+
+2 – A stateful app, wants replicas distributed in multiple clusters, such that it can form multiple smaller sized local clusters using only local replicas.
+
```

Comment:
What about two stateful sets that need to comminicate with each other. Many applications such as Cassandra, Kafka and elasticsearch include the capability to span physical data centers?